src/input.service.ts

```diff
@@ -289,78 +230,55 @@ export class InputService {
-    ]],
-    [Direction.DOWN, [
-      40,  // UpArrow
-      212, // GamepadLeftThumbstickDown
-      204, // GamepadDPadDown
-      139, // NavigationDown
-    ]],
-    [Direction.SUBMIT, [
-      13,  // Enter
-      32,  // Space
-      142, // NavigationAccept
-      195, // GamepadA
-    ]],
-    [Direction.BACK, [
-      8,   // Backspace
-      196, // GamepadB
-    ]],
-    [Direction.X, [
-      103, // Numpad 7
-      197, // GamepadX
-    ]],
-    [Direction.Y, [
-      105,   // Numpad 9
-      198, // GamepadY
-    ]],
-    [Direction.TABLEFT, [
-      100, // Numbpad Left
-      200, // Left Bumper
-    ]],
-    [Direction.TABRIGHT, [
-      102, // Numpad Right
-      199, // Right Bumper
-    ]],
-    [Direction.TABUP, [
-      104, // Numpad Up
-      201, // Left Trigger
-    ]],
-    [Direction.TABDOWN, [
-      98, // Numpad Down
-      202, // Right Trigger
-    ]],
-    [Direction.VIEW, [
-      111, // Numpad Divide
-      208, // View Button
-    ]],
-    [Direction.MENU, [
-      106, // Numpad Multiply
-      207, // Menu Button
-    ]],
+  public codeDirectionMap = new Map<number, Direction>([
+    [37, Direction.LEFT],      // LeftArrow
+    [214, Direction.LEFT],     // GamepadLeftThumbstickLeft
+    [205, Direction.LEFT],     // GamepadDPadLeft
+    [140, Direction.LEFT],     // NavigationLeft
+    [39, Direction.RIGHT],     // RightArrow
+    [213, Direction.RIGHT],    // GamepadLeftThumbstickRight
+    [206, Direction.RIGHT],    // GamepadDPadRight
+    [141, Direction.RIGHT],    // NavigationRight
+    [38, Direction.UP],        // UpArrow
+    [211, Direction.UP],       // GamepadLeftThumbstickUp
+    [203, Direction.UP],       // GamepadDPadUp
+    [138, Direction.UP],       // NavigationUp
+    [40, Direction.DOWN],      // DownArrow
+    [212, Direction.DOWN],     // GamepadLeftThumbstickDown
+    [204, Direction.DOWN],     // GamepadDPadDown
+    [139, Direction.DOWN],     // NavigationDown
+    [13, Direction.SUBMIT],    // Enter
+    [32, Direction.SUBMIT],    // Space
+    [142, Direction.SUBMIT],   // NavigationAccept
+    [195, Direction.SUBMIT],   // GamepadA
+    [8, Direction.BACK],       // Backspace
+    [196, Direction.BACK],     // GamepadB
+    [103, Direction.X],        // Numpad 7
+    [197, Direction.X],        // GamepadX
+    [105, Direction.Y],        // Numpad 9
+    [198, Direction.Y],        // GamepadY
+    [100, Direction.TABLEFT],  // Numbpad Left
+    [200, Direction.TABLEFT],  // Left Bumper
+    [102, Direction.TABRIGHT], // Numpad Right
+    [199, Direction.TABRIGHT], // Right Bumper
+    [104, Direction.TABUP],    // Numpad Up
+    [201, Direction.TABUP],    // Left Trigger
+    [98, Direction.TABDOWN],   // Numpad Down
+    [202, Direction.TABDOWN],  // Right Trigger
+    [111, Direction.VIEW],     // Numpad Divide
+    [208, Direction.VIEW],     // View Button
+    [106, Direction.MENU],     // Numpad Multiply
+    [207, Direction.MENU],     // Menu Button
```

Comment:
While this is true, do note that maps in JavaScript are still hella fast and should not be avoided simply for want of more perf in cases where they would be cleaner/clearer.

```js
/*
      34,838,996 op/s » map
      45,918,368 op/s » array

  Suites:  0
  Benches: 2
  Elapsed: 1,579.82 ms
*/

const map = { '1': 1, '500': 2, '1000': 3 };
const arr = [];
arr[1] = 1;
arr[500] = 2;
arr[1000] = 3;

let x = 0;
bench('map', () => x ^= map['1'] + map['500'] + map['1000'])
bench('array', () => x ^= arr[1] + arr[500] + arr[1000])
```